src/util.js

```diff
@@ -55,29 +25,42 @@ exports.serialize = (dagNode, callback) => {
   callback(null, serialized)
 }
 
-exports.deserialize = (data, callback) => {
+function deserialize (data, callback) {
   const pbn = proto.PBNode.decode(data)
 
   const links = pbn.Links.map((link) => {
     return new DAGLink(link.Name, link.Tsize, link.Hash)
   })
 
-  stable.inplace(links, exports.linkSort)
-
   const buf = pbn.Data || new Buffer(0)
 
-  const dagNode = new DAGNode(buf, links)
-
-  callback(null, dagNode)
+  DAGNode.create(buf, links, callback)
```

Comment:
We could break util into separate files for each function, avoiding that 'cyclic' dependency, do note that in CJS, modules loaded are cached, solving this issue without any intervention.

Nevertheless, this is not a IPLD Format issue, if an issue at all, is just because we want the DAGNode instances to be immutable and easy to work with and so, we need the serialize version upon its creation.